packages/manager/apps/dedicated/client/app/license/order/order-legacy/controller.js

```diff
@@ -0,0 +1,360 @@
+    if (
+      type &&
+      type !== this.licenseType &&
+      this.isAvailable(type) &&
+      !this.loaders.prices
+    ) {
+      // In case of license upgrade, show an information popup and redirect to upgrade screen.
+      const existingServiceName = this.getExistingServiceName(type);
+      if (existingServiceName) {
+        this.setAction({
+          action: 'redirection/upgrade/license-redirection-upgrade',
+          data: {
+            licenseId: existingServiceName,
+            licenseType: type,
+          },
+        });
+        return;
+      }
+
+      this.licenseType = type;
+      this.version = null;
+      if (
+        this.types[this.licenseType].options?.length > 0 &&
+        this.types[this.licenseType].options[0].options?.length === 1
+      ) {
+        // eslint-disable-next-line prefer-destructuring
+        this.version = this.types[this.licenseType].options[0].options[0];
+      }
+      const { options } = this.types[this.licenseType].options[0] || [];
+      if (this.licenseType === LICENCE_TYPES.CPANEL) {
+        this.formatedOptions = options.map((option) => {
+          const formattedOption = {
+            ...option,
+          };
+          const value = formattedOption.value || '';
+          switch (true) {
+            case value.includes(CPANEL_SOLO_TYPE):
+              formattedOption.cPanelCount = MAX_CPANEL_SOLO_ACCOUNTS;
+              formattedOption.displayName = this.$translate.instant(
+                'license_designation_CPANEL_solo_option',
+              );
+              break;
+            case CPANEL_PREMIER_TYPES.filter((cPanelType) =>
+              value.includes(cPanelType),
+            ).length > 0:
+              formattedOption.cPanelCount = parseInt(
+                head(value.match(/[\d]+/)),
+                10,
+              );
+              formattedOption.displayName = this.$translate.instant(
+                'license_designation_CPANEL_premier_option',
+                { numberOfCPANEL: formattedOption.cPanelCount },
+              );
+              break;
+            case value.includes(CPANEL_PLUS_TYPE):
+              formattedOption.cPanelCount = MAX_CPANEL_PLUS_ACCOUNTS;
+              formattedOption.displayName = this.$translate.instant(
+                'license_designation_CPANEL_plus_option',
+              );
+              break;
+            case value.includes(CPANEL_ADMIN_TYPE):
+              formattedOption.cPanelCount = MAX_CPANEL_ADMIN_ACCOUNTS;
+              formattedOption.displayName = this.$translate.instant(
+                'license_designation_CPANEL_admin_option',
+              );
+              break;
+            case value.includes(CPANEL_PRO_TYPE):
+              formattedOption.cPanelCount = MAX_CPANEL_PRO_ACCOUNTS;
+              formattedOption.displayName = this.$translate.instant(
+                'license_designation_CPANEL_pro_option',
+              );
+              break;
+            default:
+              break;
```

Comment:
```suggestion
```
No need if you don't have default case.

Reply:
I'll try, but I think lint will force me to have a default block